client/src/components/ProfileEdit.jsx

```diff
@@ -6,93 +6,143 @@ import { Link } from 'react-router-dom'
-	}
+    state = {
+        user_edit: {},
+        error_nick: "",
+        error_repeated: ""
+    }
 
-	componentDidMount() {
-		request('/profile/' + this.props.match.params.profile_id, "GET")
-    .then(json => {
-			this.setState({user_edit: json});
-      console.log(this.state.user_edit)
-      this.setState({
-        user_edit: {
-          f_name: this.state.user_edit.first_name,
-          l_name: this.state.user_edit.last_name,
-          nick: this.state.user_edit.nickname,
-          loc: this.state.user_edit.location,
-          pass: this.state.user_edit.password
-        }
-      })
-		})
-	}
+    componentDidMount() {
+        request('/profile/' + this.props.match.params.profile_id)
+        .then(json => {
+            this.setState({user_edit: json});
+            this.setState({
+                user_edit: {
+                    f_name: this.state.user_edit.first_name,
+                    l_name: this.state.user_edit.last_name,
+                    nick: this.state.user_edit.nickname,
+                    loc: this.state.user_edit.location,
+                    pass: this.state.user_edit.password
+                }
+            })
+        })
+    }
 
-	UpdateClick = (eve) => {
-		eve.preventDefault()
+    UpdateClick = (eve) => {
+        eve.preventDefault()
 
-    let nick_error = ""
-    let repeated_error = ""
+        let msg_rep = ""
 
-			if (this.state.new_password === this.state.repeat_password) {
-				const data = {
-					"first_name": this.state.f_name,
-					"last_name": this.state.l_name,
-					"nickname": this.state.nickname,
-					"location": this.state.location,
-					"password": this.state.new_password,
-					"id": this.state.user_edit.id
-				}
-				request('/profile/' + this.props.match.params.profile_id, "PUT", JSON.stringify(data))
-        .then(data => {
-          if (data.msg !== "") {
-            this.setState({ error_nick: data.msg })
-          }
-          else {
-            this.props.history.push('/profile/' + this.props.match.params.profile_id);
-            window.Materialize.toast("Profile Updated", 2500);
-          }
-        })
-			} else {
-        repeated_error = "Incorrect repeated password"
-			}
-      this.setState({ error_repeated: repeated_error })
-	};
+        this.setState({ error_repeated: ""})
+
+        if (this.state.new_password === this.state.repeat_password) {
+            const data = {
+                "first_name": this.state.f_name,
+                "last_name": this.state.l_name,
+                "nickname": this.state.nickname,
+                "location": this.state.location,
+                "password": this.state.new_password,
+                "id": this.state.user_edit.id
+            }
+            request('/profile/' + this.props.match.params.profile_id, "PUT", JSON.stringify(data))
+            .then(data => {
+                if (data.msg !== "") {
+                    this.setState({ error_nick: data.msg })
+                }
+                else {
```

Comment:
You are using 'Colander', so if data will be invalid, it returns JSON with `error` field and `if (data.msg !== "")` will return true but you don't handle this error.